src/NHibernate/Cfg/MappingSchema/HbmOneToOne.cs

```diff
@@ -18,9 +18,13 @@ public string Access
 			get { return access; }
 		}
 
+		// 6.0 Todo : remove XmlIgnore after removing the setter. See #3607 fix.
+		[XmlIgnore]
 		public bool OptimisticLock
 		{
 			get => optimisticlock;
+			// Since v5.4.10
+			[Obsolete("Providing a setter for OptimisticLock was unintended and will be removed.")]
 			set => optimisticlock = value;
```

Comment:
Providing a setter cause the property to be serialized to XML, which causes #3607. And there are no usage for a setter here.

Reply:
@bahusoid, that setter was added in #3216. Was there any reason for it I am missing?

Reply:
Nope. It's some mistake.

Reply:
Ok, thanks.